lang/python/homeassistant/patches/0001-Update-dependencies-to-their-latest-version.patch

```diff
@@ -0,0 +1,81 @@
+From fbaecea5672f5ea687d14f01e654b1c22c1f3c15 Mon Sep 17 00:00:00 2001
+From: Josef Schlehofer <pepe.schlehofer@gmail.com>
+Date: Sun, 7 Jul 2019 08:49:31 +0200
+Subject: [PATCH] Update dependencies to their latest version
+
+The latest version of Home Assistant 0.95.4 doesn't work for me.
+See for more details https://github.com/home-assistant/home-assistant/issues/24982
+
+Home Assistant uses pinned versions of dependencies, let's change them to
+their latest version as they are already included in newer versions.
+I'm thinking as it was suggested by jefferyto a in pull request to
+include it in OpenWrt:
+https://github.com/openwrt/packages/pull/8750#issuecomment-485938034
+to remove entire the pinning version
+---
+ homeassistant/package_constraints.txt | 14 +++++++-------
+ setup.py                              | 14 +++++++-------
+ 2 files changed, 14 insertions(+), 14 deletions(-)
+
+diff --git a/homeassistant/package_constraints.txt b/homeassistant/package_constraints.txt
+index a4a08af12..bf72b2825 100644
+--- a/homeassistant/package_constraints.txt
++++ b/homeassistant/package_constraints.txt
+@@ -2,17 +2,17 @@ aiohttp==3.5.4
+ astral==1.10.1
+ async_timeout==3.0.1
+ attrs==19.1.0
+-bcrypt==3.1.6
+-certifi>=2018.04.16
+-jinja2>=2.10
++bcrypt==3.1.7
++certifi>=2019.06.16
++jinja2>=2.10.1
```

Comment:
If you are talking about file package_constraints.txt, I can drop my patch for it and see if it works, but I did that when I was doing pull requests to Home Assistant, so I just cherry-picked the commit and included a commit message and I just continue with that. 

About the dependencies in Home Assistant. 
I need to patch them all otherwise it won't work if I had installed an old version of some dependencies. It throws me this error:

> pkg_resources.DistributionNotFound: The 'ruamel.yaml==0.15.97' distribution was not found and is required by homeassistant

However, I'd like to get rid of this patch as much as possible. I'm trying to track changes for dependencies in Hass and here, but the latest version of Hass isn't working for me. I created an issue in their Github repository and in the meantime, I ended up to get to run an old version, which worked for me in the past. I know, it's not perfect right now, but I'd like to finish this PR.

Reply:
I know but version 2.10.1 is valid for both `jinja2>=2.10` and `jinja2>=2.10.1`.

I am thinking if it would not make sense to just drop whole REQUIRES variable instead of this patching. I would do something like `sed -i '/^REQUIRES = [/,/^]$/{/^ /d};' setup.py`

It is not ideal because there is possibility to miss new packages so it requires manual work as well but I think that it should just work.

Another option is to replace in general all pinning with request on newer version:
`sed -i '/^REQUIRES = [/,/^]$/{s/==/>=/};' setup.py` (this requires more tweaking to also remove upper bounds)

Reply:
Yes, that's what @miska suggested me yesterday, while I told him that I'm not friend with **sed**, because right now with this solution I need to edit the patch everytime when some dependency changes. I will edit it.